test/integration/common_test.go

```diff
@@ -39,6 +40,13 @@ func random_domain() string {
 	return hex.EncodeToString(bytes[:]) + ".com"
 }
 
+func random_ipv6() string {
```

Comment:
I don't see any callers for this function. How is this passing the dead-code check? Oh no, I bet the linter doesn't run with the +integration build tag so it ignores this file...